### ISSUE TITLE
Stopped consumers are not reported because of rule 1

### ISSUE THREAD

**Issue by rconn01** [linkedin/Burrow#290]
If we take the following scenario it appears that the consumer will never report as stopped.

1. Group A is consuming and a full window is built will all zeros
1. Group A stops consuming
1. Messages produce to partitions that are part of group A
1. The total lag for Group A increases

I would think that at some point rule 3 would kick in. The difference between the first and last offset in the window is less than now and the last offset. The problem is the window is filled with all `0s` so rule 1 wins and the group has a status of `ok` forever

https://github.com/linkedin/Burrow/blob/b531ffdd9ef9835ce56bb2df068c75fbda36e60e/core/internal/evaluator/caching.go#L304

**Comment by toddpalino** [comment]
currentLag is calculated based on the broker offset at the time of the evaluation request, and the last consumer offset. So if messages are produced to the partition after the last consumer offset commit, currentLag should be positive and non-zero:

from inmemory.go:
```
		for p, partition := range partitions {
			if len(partition.Offsets) > 0 {
				brokerOffset := topicMap[p].Offset
				lastOffset := partition.Offsets[len(partition.Offsets)-1]
				if lastOffset != nil {
					if brokerOffset < lastOffset.Offset {
						// Little bit of a hack - because we only get broker offsets periodically, it's possible the consumer offset could be ahead of where we think the broker
						// is. In this case, just mark it as zero lag.
						partition.CurrentLag = 0
					} else {
						partition.CurrentLag = uint64(brokerOffset - lastOffset.Offset)
					}
				}
			}
		}
```

If you have a specific case where this is happening, you can use the API to dump the full consumer information for the group and we can check out the information that's there.

**Comment by rconn01** [comment]
Agree that currentLag is positive and non zero. I think the issue is the fact that the window has a lag of zero in it so `isLagAlwaysNotZero()` returns `false`. Which makes status always remain `ok`

Here is the output of the `/status` endpoint showing the case

```json
{
   "error":false,
   "message":"consumer status returned",
   "status":{
      "cluster":"local",
      "group":"lag-test-group",
      "status":"OK",
      "complete":1,
      "partitions":[

      ],
      "partition_count":1,
      "maxlag":{
         "topic":"ray_test",
         "partition":0,
         "owner":"",
         "status":"OK",
         "start":{
            "offset":792748079,
            "timestamp":1512224618356,
            "lag":0
         },
         "end":{
            "offset":792748093,
            "timestamp":1512224632397,
            "lag":0
         },
         "current_lag":202,
         "complete":1
      },
      "totallag":202
   },
   "request":{
      "url":"/v3/kafka/local/consumer/lag-test-group/status",
      "host":""
   }
}
``` 

and the output for the consumer endpoint

```json
{
   "error":false,
   "message":"consumer detail returned",
   "topics":{
      "ray_test":[
         {
            "offsets":[
               {
                  "offset":792748079,
                  "timestamp":1512224618356,
                  "lag":0
               },
               {
                  "offset":792748080,
                  "timestamp":1512224619362,
                  "lag":0
               },
               {
                  "offset":792748081,
                  "timestamp":1512224620366,
                  "lag":0
               },
               {
                  "offset":792748082,
                  "timestamp":1512224621367,
                  "lag":0
               },
               {
                  "offset":792748083,
                  "timestamp":1512224622370,
                  "lag":0
               },
               {
                  "offset":792748084,
                  "timestamp":1512224623373,
                  "lag":0
               },
               {
                  "offset":792748085,
                  "timestamp":1512224624378,
                  "lag":0
               },
               {
                  "offset":792748086,
                  "timestamp":1512224625379,
                  "lag":0
               },
               {
                  "offset":792748087,
                  "timestamp":1512224626383,
                  "lag":0
               },
               {
                  "offset":792748088,
                  "timestamp":1512224627383,
                  "lag":0
               },
               {
                  "offset":792748089,
                  "timestamp":1512224628383,
                  "lag":0
               },
               {
                  "offset":792748090,
                  "timestamp":1512224629388,
                  "lag":0
               },
               {
                  "offset":792748091,
                  "timestamp":1512224630391,
                  "lag":0
               },
               {
                  "offset":792748092,
                  "timestamp":1512224631394,
                  "lag":0
               },
               {
                  "offset":792748093,
                  "timestamp":1512224632397,
                  "lag":0
               }
            ],
            "owner":"",
            "current-lag":391
         }
      ]
   },
   "request":{
      "url":"/v3/kafka/local/consumer/lag-test-group",
      "host":""
   }
}
```

**Comment by toddpalino** [comment]
Ah, I understand what you're saying now. Thanks for providing the consumer detail to look at. I'm looking at the tests and the code now, and I think I see where the flaw is. There's actually a test that should catch this, but the comment and the outcome is set wrong. It tests like this is intended behavior, and it's definitely not when it comes to stop. For stall, it's OK to wait for the zeros to clear out (since the consumer is still committing).

I just created PR #292 to address this flaw, and I've added a test with your specific offset information to make sure it gives the right answer. Take a look and tell me what you think.
